pandas/core/indexing.py

```diff
@@ -2117,7 +2117,8 @@ def _setitem_with_indexer_missing(self, indexer, value):
                 curr_dtype = getattr(curr_dtype, "numpy_dtype", curr_dtype)
                 new_dtype = maybe_promote(curr_dtype, value)[0]
             else:
-                new_dtype = None
+                if isinstance(self.obj.dtype, object):
```

Comment:
I don't think this is a good idea. This breaks the current behavior and the special case for object looks a bit weird. What happens if you assign a different dtype? This would still change which is inconsistent then-

Also you have to keep the fall through clause

Reply:
also wont this isinstance be True for _everything_?

Reply:
> also wont this isinstance be True for _everything_? yes

Reply:
> I don't think this is a good idea. This breaks the current behavior and the special case for object looks a bit weird. What happens if you assign a different dtype? This would still change which is inconsistent then-
> 
> Also you have to keep the fall through clause

From this issue https://github.com/pandas-dev/pandas/issues/19647 , when adding element to empty object Series. The series should change based on type object which is set by user.

```
s = pd.Series(dtype='object')
s.loc['myint'] = 1
s.loc['myfloat'] = 2.
```

Expected output:
```
myint      1.0
myfloat    2.0
dtype: object
```

If I assign to different dtype,  for example
```
s = pd.Series(dtype=float)
s.loc['myint'] = 1.0
s.loc['myfloat'] = 2
```
Output:
```
myint      1.0
myfloat    2.0
dtype: float64
```

Reply:
What happens if the empty series has some other dtype apart from object? Then this is not caught by your current change.

on the other side, if the dtype is the default this should still change, this is not covered here

Reply:
> What happens if the empty series has some other dtype apart from object? Then this is not caught by your current change.
If the empty series has some other dtype apart from object, for example 
```
s = pd.Series(dtype=float)
s.loc['myint'] = 1.0
s.loc['myfloat'] = 2
print(s)
```
Output will be like this
```myint      1.0
myfloat    2.0
dtype: float64
```